components/console-backend-service/internal/domain/assetstore/clusterasset_converter.go

```diff
@@ -23,11 +27,16 @@ func (c *clusterAssetConverter) ToGQL(item *v1alpha2.ClusterAsset) (*gqlschema.C
 	}
 
 	status := c.extractor.Status(item.Status.CommonAssetStatus)
+	metadata, err := c.extractMetadata(item.Spec.Metadata)
```

Comment:
Please change name of type of `extractor` from `AssetStatusExtractor` to `AssetExtractor` and put there `extractMetadata` and use it here.